lib/dialects/abstract/query-generator.js

```diff
@@ -1666,6 +1703,240 @@ const QueryGenerator = {
     };
   },
 
+  generateCTEQuery(topLevelInfo) {
```

Comment:
These methods just deal with CTE, we should extract them to a different file, may be `abstract/cte.js` or `abstract/generators/cte.js`

They can then extend on abstract class prototype

Reply:
There is no need to abstraction here IMHO because we can still overload the CTE methods in the dialect query-generator if need be.

Reply:
This is what is holding up the getting the PR finished, mainly because I have not had tie to address it. The CTE functions are pretty tightly coupled with the rest of QueryGenerator functions to the point where there is not an obvious way to move them to another file. I have not had the opportunity to sit down and give it a hard look, however.